File: api/src/main/java/com/ruby/rubia_server/core/entity/MessageTemplate.java
package com.ruby.rubia_server.core.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Entity
@Table(name = "message_templates")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class MessageTemplate {

    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    private UUID id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "company_id", nullable = false)
    private Company company;

    @Column(nullable = false)
    private String name;

    @Column(columnDefinition = "TEXT", nullable = false)
    private String content;

    @Column(name = "is_ai_generated", nullable = false)
    @Builder.Default
    private Boolean isAiGenerated = false;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by_user_id")
    private User createdBy;

    // Novo campo: Qual agente de IA gerou este template (se aplicável)
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "ai_agent_id")
    private AIAgent aiAgent;

    // Novo campo: Tom/estilo do template
    @Column(name = "tone")
    private String tone; // Ex: "FORMAL", "INFORMAL", "DESCONTRAIDO", "EMPATICO"

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "last_edited_by_user_id")
    private User lastEditedBy;

    @Column(name = "edit_count")
    @Builder.Default
    private Integer editCount = 0;

    @OneToMany(mappedBy = "template", cascade = CascadeType.ALL, orphanRemoval = true, fetch = FetchType.LAZY)
    @OrderBy("revisionNumber DESC")
    private List<MessageTemplateRevision> revisions;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}

File: api-dotnet/Rubia.Server/Entities/MessageTemplate.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Rubia.Server.Entities;

[Table("message_templates")]
public class MessageTemplate : BaseEntity
{
    [Required]
    [Column("name")]
    public string Name { get; set; } = string.Empty;

    [Required]
    [Column("content", TypeName = "TEXT")]
    public string Content { get; set; } = string.Empty;

    [Column("is_ai_generated")]
    [Required]
    public bool IsAiGenerated { get; set; } = false;

    [Column("tone")]
    public string? Tone { get; set; } // Ex: "FORMAL", "INFORMAL", "DESCONTRAIDO", "EMPATICO"

    [Column("edit_count")]
    public int EditCount { get; set; } = 0;

    [Column("deleted_at")]
    public DateTime? DeletedAt { get; set; }

    // Navigation properties
    [Column("company_id")]
    [Required]
    public Guid CompanyId { get; set; }
    
    [ForeignKey("CompanyId")]
    public virtual Company Company { get; set; } = null!;

    [Column("created_by_user_id")]
    public Guid? CreatedByUserId { get; set; }
    
    [ForeignKey("CreatedByUserId")]
    public virtual User? CreatedBy { get; set; }

    [Column("ai_agent_id")]
    public Guid? AiAgentId { get; set; }
    
    [ForeignKey("AiAgentId")]
    public virtual AIAgent? AiAgent { get; set; } // Qual agente de IA gerou este template (se aplicável)

    [Column("last_edited_by_user_id")]
    public Guid? LastEditedByUserId { get; set; }
    
    [ForeignKey("LastEditedByUserId")]
    public virtual User? LastEditedBy { get; set; }

    public virtual ICollection<MessageTemplateRevision> Revisions { get; set; } = new List<MessageTemplateRevision>();
    public virtual ICollection<Message> Messages { get; set; } = new List<Message>();
    public virtual ICollection<Campaign> CampaignsAsInitialTemplate { get; set; } = new List<Campaign>();
}